### PR TITLE
Have Windows and VMS build files use shlib_variant

### LINKED ISSUE #20732: How to rename output dll & libs of openssl  1.1.1t on windows
I am using openssl 1.1.1t,  I just want to rename the out dll, like libssl-1_1.lib->libssl-test-1_1.lib, libcrypto-1_1.lib->libcrypto-test-1_1.lib, I have refer the issue #20439 , but it seems the solution did not work on windows. So do we havn another solution to do this, thanks

### PR DESCRIPTION
**Pull request by levitte**
This is an omission, it should have been in place a long time ago.

Fixes #20732

### CODE REVIEW

File: Configurations/descrip.mms.tmpl [hunk 1]
```diff
@@ -47,13 +47,13 @@
       map { (my $x = $_) =~ s/\.a$//; $x }
       @{$unified_info{libraries}};
   our @shlibs =
-      map { $unified_info{sharednames}->{$_} || () }
+      map { $unified_info{sharednames}->{$_}.$shlibvariant || () }
```
Comment (tom-cosgrove-arm) [inline Configurations/descrip.mms.tmpl:50]: Sorry for the stupid question, but where does `$shlibvariant` come from? I see

`sub shlibvariant        { $target{shlib_variant} || "" } `

but presumably there's an `eval` somewhere that creates the scalar?

Comment (levitte) [reply Configurations/descrip.mms.tmpl:52]: Perfectly valid question!  I had forgotten to define that variable.  Fixed

Comment (tom-cosgrove-arm) [reply Configurations/descrip.mms.tmpl:52]: Ah, the perils of legacy Perl code without `use strict`!

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Comment by t8m** [comment]
The PR should be targetted against master branch. IMO this should not go into 1.1.1.

**Comment by tom-cosgrove-arm** [comment]
@t8m is this okay to go into 3.0 and 3.1 branches?

**Comment by t8m** [comment]
> @t8m is this okay to go into 3.0 and 3.1 branches?

IMO it is slightly borderline but OK to go there.

**Comment by levitte** [comment]
Er.....  this was made against the 1.1.1 branch for a reason!  This fix isn't at all necessary in 3.0, 3.1 or in the master branch.
That should be clear if you have a look at #20732.

**Comment by levitte** [comment]
FYI: with 3.0 and on, a lot of the platform specific stuff was moved from the build file templates to `Configurations/platform.pm` and associated files in `Configurations/platform/`.  You'll find that `shlib_variant` is consistently used across all supported platform families.  Therefore, this PR is not for those branches.

**Comment by t8m** [comment]
Actually we said bugfixes can be merged if they appear as PR, we are just not committed to fixing bugs on 1.1.1. So IMO this can go in 1.1.1.

**Review by t8m — APPROVED** [review]

**Comment by levitte** [comment]
@nickychen123456, please try this?

**Comment by hlandau** [comment]
Merged to 1.1.1. Thank you.

**Pull request closed without merging**